test/unit/spec/MfaVerify_spec.js

```diff
@@ -2301,6 +2305,76 @@ function (Okta,
+                test.setNextResponse(resChallengePush);
+                test.form[0].submit();
+                return tick(test).then(function() {
+                  expect(LoginUtil.callAfterTimeout.calls.argsFor(1)[1]).toBe(6000);
+                  var deferred = Q.defer();
+                  expect(test.router.controller.model.appState.get('transaction').status).toBe('MFA_CHALLENGE');
+                  var transaction = test.router.controller.model.appState.get('transaction');
+                  spyOn(transaction, 'poll');
+                  expect(transaction.poll).not.toHaveBeenCalled();
+                  setTimeout(deferred.resolve, 150);
+                  return deferred.promise.then(function() {
+                    expect(transaction.poll).toHaveBeenCalled();
+                  });
+                });
+              }); 
+            });
+            itp('does not start poll if factor was switched before 6000ms', function () {
+              return setupOktaPush().then(function (test) {
+                spyOn(LoginUtil, 'callAfterTimeout').and.callFake(function() {
+                  // reducing the timeout to 100 so that test is fast.
+                  return setTimeout(arguments[0], 100);
+                });
+                $.ajax.calls.reset();
+                test.setNextResponse(resChallengePush);
+                test.form[0].submit();
+                return tick(test).then(function() {
+                  var deferred = Q.defer();
+                  expect(test.router.controller.model.appState.get('transaction').status).toBe('MFA_CHALLENGE');
+                  var transaction = test.router.controller.model.appState.get('transaction');
+                  spyOn(transaction, 'poll');
+                  test.router.controller.model.appState.trigger('factorSwitched');
+                  setTimeout(deferred.resolve, 150);
+                  return deferred.promise.then(function() {
+                    expect(transaction.poll).not.toHaveBeenCalled();
+                  });
+                });
+              });
+            });
+            itp('stops listening on factorSwitched when we start polling', function () {
+              return setupOktaPush().then(function (test) {
+                return setupPolling(test, resRejectedPush)
+                .then(function () { return tick(test); }) // Final response - REJECTED
+                .then(function (test) {
+                  expect(test.form.errorMessage()).toBe('You have chosen to reject this login.');
+                  expect(test.form.submitButton().prop('disabled')).toBe(false);
+
+                  // Setup responses
+                  $.ajax.calls.reset();
+                  test.setNextResponse([resChallengePush, resChallengePush, resSuccess]);
+
+                  // Click submit
+                  test.form.submit();
+                  spyOn(test.router.controller.model, 'stopListening').and.callThrough();
+                  return tick()
```

Comment:
instead of all this `ticks`, can we do something similar to the above tests?

Reply:
Yeah I guess more ticks are not needed here. Modified the test.